### DISCUSSION TITLE
migrations for upcoming changes to reusable test suites

### DISCUSSION THREAD

**Discussion by boneskull** [appium/appium#17205] [General]
**If you are working on an extension for Appium 2, this may apply to you.**

PR #16907 relocates the reusable test suites from `@appium/base-driver` and helpers from `@appium/base-plugin` into the `@appium/test-support` package.  This will be merged and published Soon™️.

_If you are using_:

- `baseDriverE2ETests()` or `baseDriverUnitTests()` from `@appium/base-driver/build/test/basedriver`, and/or
- `e2eSetup()` from `@appium/base-plugin/build/test/helpers`

... you will need to change how you import/require these functions.  See below.

## Migrations

_These migrations assume you already have set a peer dependency and dev dependency upon Appium._

### TL;DR

For drivers:

- `baseDriverE2ETests()` renamed to `driverE2ETestSuite()`
- `baseDriverUnitTests()` renamed to `driverUnitTestSuite()`

For plugins:

- `e2eSetup()` renamed to `pluginE2EHarness()`

**Import all of these from `appium/test` instead of the current modules.**

### Migration for `baseDriverE2ETests`

Change:

```js
import {baseDriverE2ETests} from '@appium/base-driver/build/test/basedriver';
// or
const {baseDriverE2ETests} = require('@appium/base-driver/build/test/basedriver');
```

to:

```js
import {driverE2ETestSuite as baseDriverE2ETests} from 'appium/test';
// or
const {driverE2ETestSuite: baseDriverE2ETests} = require('appium/test');
```

### Migration for `baseDriverUnitTests`

Change:

```js
import {baseDriverUnitTests} from '@appium/base-driver/build/test/basedriver';
// or
const {baseDriverUnitTests} = require('@appium/base-driver/build/test/basedriver');
```

to:

```js
import {driverUnitTestSuite as baseDriverUnitTests} from 'appium/test';
// or
const {driverUnitTestSuite: baseDriverUnitTests} = require('appium/test');
```

### Migration for `e2eSetup`

Change

```js
import {e2eSetup} from '@appium/base-plugin/build/test/helpers';
// or
const {e2eSetup} = require('@appium/base-plugin/build/test/helpers');
```

to:

```js
import {pluginE2EHarness as e2eSetup} from 'appium/test';
// or
const {pluginE2EHarness: e2eSetup} = require('appium/test');
```
